### PR TITLE
[#1118] --  Converted get_location.js to react component

### PR DESCRIPTION
**Pull request by jm96441n**
# Description 

Convert `get_location.js` to be a react component

## Corresponding Issue

https://github.com/ifmeorg/ifme/issues/1118

# Screenshots

N/A

# Test Coverage

✅

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -48,9 +48,8 @@
 
     <% if join_path? || update_account_path? %>
       <!-- Google Maps API -->
-      <% google_maps_url = "https://maps.googleapis.com/maps/api/js?key=#{ENV['GOOGLE_API_KEY']}&libraries=places&callback=initAutocomplete" %>
-      <link rel="preconnect" href=<%= google_maps_url %>>
-      <%= javascript_include_tag google_maps_url, async: true %>
+      <% google_maps_url = "https://maps.googleapis.com/maps/api/js?key=#{ENV['GOOGLE_API_KEY']}&libraries=places" %>
```
Comment (jm96441n) [inline app/views/layouts/application.html.erb]: If it's preferred I can add this to the `InputLocation` component and lazy load it there, but I think the performance will be better with this load happening

Comment (jm96441n) [reply app/views/layouts/application.html.erb]: found a potentially better way of doing this by loading the google maps scripts via yarn, trying this out now

File: app/views/devise/registrations/edit.html.erb [hunk 2]
```diff
@@ -5,7 +5,7 @@
   <%= f.text_field :name, 'aria-label': t('common.name') %>
 
   <div class="label smallMarginTop"><%= t('common.form.location') %></div>
-  <%= f.text_field :location, placeholder: t('devise.registrations.edit.placeholders.optional'), 'aria-label': t('common.form.location') %>
+  <%= react_component 'InputLocation', props: { placeholder: t('devise.registrations.edit.placeholders.optional') } %>
```
Comment (julianguyen) [inline app/views/devise/registrations/edit.html.erb]: Just for consistency we actually use `InputLocation` via the generic `Input` component. I would specify the type prop as `location`.

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (julianguyen) [reply app/views/layouts/application.html.erb]: If there is a better way to do this via yarn, we should also add this component to Storybook as well

File: client/app/components/Input/InputLocation.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    const { id } = this.props;
+    const { address } = this.state;
+    return (
+      <LocationAutocomplete
+        name="user[location]"
+        placeholder={placeholder}
+        googleAPIKey={apiKey}
+        onChange={this.handleChange}
+        onDropdownSelect={this.handleDropdownSelect}
+        className="smallerMarginBottom"
```
Comment (julianguyen) [inline client/app/components/Input/InputLocation.jsx]: This class name no longer exists actually after a recent CSS refactor. You can use `smallMarginBottom` or set custom css.

File: config/webpacker.yml [hunk 4]
```diff
@@ -24,6 +24,9 @@ development:
 test:
   <<: *default
 
+  # Tell webpack where the app is located
+  source_path: client/app
```
[the remark this answers is not in the document]
Comment (julianguyen) [reply config/webpacker.yml]: Woo glad you figured this out, thank you!

**Comment by julianguyen** [comment]
No problem, thanks for the update! Good luck on midterms!

**Comment by julianguyen** [comment]
Hey @jm96441n, just following up with this! Hope things are decent :)

**Comment by julianguyen** [comment]
I'm just going to wrap up this PR. I hope things are decent on your end :)

File: app/views/devise/registrations/edit.html.erb [hunk 5]
```diff
@@ -5,7 +5,7 @@
   <%= f.text_field :name, 'aria-label': t('common.name') %>
 
   <div class="label smallMarginTop"><%= t('common.form.location') %></div>
-  <%= f.text_field :location, placeholder: t('devise.registrations.edit.placeholders.optional'), 'aria-label': t('common.form.location') %>
+  <%= react_component 'Input', props: { type: 'location', placeholder: t('devise.registrations.edit.placeholders.optional'), googleAPIKey: ENV['GOOGLE_API_KEY'], id: 'user_input'} %>
```
Comment (julianguyen) [inline app/views/devise/registrations/edit.html.erb]: Hmm I wonder if it's safe to pass in the API key like this 🤔

Comment (julianguyen) [reply app/views/devise/registrations/edit.html.erb]: NVM, it should be fine!

**Comment by julianguyen** [comment]
LOL damn, I can't get rid of jQuery cause or else some devise functionality stops working. https://github.com/plataformatec/devise/issues/4486

**Pull request merged**